nautobot/dcim/migrations/0023_create_region_and_site_location_types_and_locations.py

```diff
@@ -0,0 +1,141 @@
+# Generated by Django 3.2.16 on 2022-12-23 02:48
+
+from django.db import migrations
+
+
+def create_region_and_site_location_types(apps, schema_editor):
+    Region = apps.get_model("dcim", "region")
+    Site = apps.get_model("dcim", "site")
+    LocationType = apps.get_model("dcim", "locationtype")
+    Location = apps.get_model("dcim", "location")
+
+    # Site and Region instances both exist
+    if Site.objects.exists() and Region.objects.exists():
+        region_lt = LocationType.objects.create(name="Region", nestable=True)
+        # Breadth First Query
+        for region in Region.objects.with_tree_fields().extra(order_by=["__tree.tree_depth", "__tree.tree_ordering"]):
+            Location.objects.create(
+                location_type=region_lt,
+                name=region.name,
+                description=region.description,
+                parent=Location.objects.get(name=region.parent.name, location_type=region_lt)
+                if region.parent
+                else None,
+            )
+        site_lt = LocationType.objects.create(name="Site", parent=LocationType.objects.get(name="Region"))
+        Location.objects.create(
+            location_type=region_lt,
+            name="Global Region",
+            description=region.description,
+        )
```

Comment:
Maybe only create the Global Region if `Site.objects.filter(region__isnull=True).exists()`?